94.md

```diff
@@ -0,0 +1,49 @@
+NIP-94 - File Header
+======
+`draft` `optional` `author:frbitten` 
+
+The purpose of this NIP is to allow an organization and classification of shared files. So that relays can filter and organize in any way that is of interest.
+Also the goal is to create a base on the protocol for this bountsr "Filesharing App" (https://bountsr.org/p2p-filesharing/) to be implemented.
+
+Nostr event
+------------------
+This NIP specifies the use of the `1063` event type, having in `content` a description of the file content, and a list of tags described below:
+* `r` the url to download the file
+* `type` a string indicating the data type of the file. The MIME types format must be used (https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types)
+* `decrypt` (optional) If the file is encrypted, you must indicate in the first value the algorithm used to encrypt the file and in the second value the parameters that allow the decryption of the file
+* `hash` containing the SHA-256 hexencoded string of the file or the BlurHash string (for images). The third array value is the lowercase chosen algorithm name: "sha256" or "blurhash".
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded sha256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <unix timestamp in seconds>,
+  "kind": 1063,
+  "tags": [
+    ["r",<string with URI of file>],
+    ["decrypt",<algorithm>,<Decryption Params>],
```

Comment:
```suggestion
    ["xchacha20", <key>, <nonce>],
```

Reply:
`["aes-256-gcm", "<key>", "<iv>", "<ts>"]`

Where `key` & `iv` are hex, `ts` is the Tagsize in bits

Reply:
https://github.com/bitwiseshiftleft/sjcl/blob/85caa53c281eeeb502310013312c775d35fe0867/core/gcm.js#L17-L24

Reply:
Or you could specify the tagsize is 128 its and just leave it out of the tag